x-pack/test/api_integration/deployment_agnostic/apis/observability/streams/queries.ts

```diff
@@ -0,0 +1,240 @@
+      const query = { id: v4(), title: 'Significant Query', kql: { query: "message:'query'" } };
+      const upsertQueryResponse = await apiClient
+        .fetch('PUT /api/streams/{name}/queries/{queryId} 2023-10-31', {
+          params: {
+            path: { name: STREAM_NAME, queryId: query.id },
+            body: {
+              title: query.title,
+              kql: query.kql,
+            },
+          },
+        })
+        .expect(200)
+        .then((res) => res.body);
+      expect(upsertQueryResponse.acknowledged).to.be(true);
+
+      const getQueriesResponse = await getQueries(apiClient, STREAM_NAME);
+      expect(getQueriesResponse.queries).to.eql([query]);
+    });
+
+    it('updates a query when already defined', async () => {
+      const queryId = v4();
+      await putStream(apiClient, STREAM_NAME, {
+        stream,
+        dashboards: [],
+        queries: [
+          {
+            id: queryId,
+            title: 'Significant Query',
+            kql: { query: "message:'query'" },
+          },
+        ],
+      });
+
+      const upsertQueryResponse = await apiClient
+        .fetch('PUT /api/streams/{name}/queries/{queryId} 2023-10-31', {
+          params: {
+            path: { name: STREAM_NAME, queryId },
+            body: {
+              title: 'Another title',
+              kql: { query: "message:'Something else'" },
+            },
+          },
+        })
+        .expect(200)
+        .then((res) => res.body);
+      expect(upsertQueryResponse.acknowledged).to.be(true);
+
+      const getQueriesResponse = await getQueries(apiClient, STREAM_NAME);
+      expect(getQueriesResponse.queries).to.eql([
+        {
+          id: queryId,
+          title: 'Another title',
+          kql: { query: "message:'Something else'" },
+        },
+      ]);
+    });
+
+    it('deletes an existing query successfully', async () => {
+      const queryId = v4();
+      await putStream(apiClient, STREAM_NAME, {
+        stream,
+        dashboards: [],
+        queries: [
+          {
+            id: queryId,
+            title: 'Significant Query',
+            kql: { query: "message:'query'" },
+          },
+        ],
+      });
+
+      const deleteQueryResponse = await apiClient
+        .fetch('DELETE /api/streams/{name}/queries/{queryId} 2023-10-31', {
+          params: { path: { name: STREAM_NAME, queryId } },
+        })
+        .expect(200)
+        .then((res) => res.body);
+      expect(deleteQueryResponse.acknowledged).to.be(true);
+
+      const getQueriesResponse = await getQueries(apiClient, STREAM_NAME);
+      expect(getQueriesResponse.queries).to.eql([]);
+    });
+
+    it('deletes an inexistant query successfully', async () => {
+      const queryId = v4();
+      const deleteQueryResponse = await apiClient
+        .fetch('DELETE /api/streams/{name}/queries/{queryId} 2023-10-31', {
+          params: { path: { name: STREAM_NAME, queryId } },
+        })
+        .expect(200)
```

Comment:
I think this should return a 404, similar to when you delete an index that does not exist

Reply:
I thought we wanted to handle them as a successful noop - will change 👍🏻